event-based-simulators/main/scripts/ExportOrImportProfileData.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+from datetime import datetime
+from os.path import isfile, join
+
+import ArgumentsAndCredentialsHandler
+
+
+def exportAllProfileData(c8y, DATA_TYPE, createFrom, createTo, filePath):
+    # Loop through the list of device in Device management
+    for device in c8y.device_inventory.select(type="c8y_EventBasedSimulator"):
+        print(f"Found device '{device.name}', id: #{device.id}, "
+              f"owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        print(f"List of {device.name}'s child devices: ")
+        for childDevice in device.child_devices:
+            print(f"Child device {childDevice.name}, id #{childDevice.id}")
+            if DATA_TYPE == "alarms":
+                listAlarms(c8y, childDevice, createFrom, createTo, filePath)
+            elif DATA_TYPE == "measurements":
+                # listing measurements of child device
+                listMeasurements(c8y, childDevice, createFrom, createTo, filePath)
+
+
+def exportSpecificProfileData(c8y, DATA_TYPE, createFrom, createTo, filePath):
+    print(f"Enter device name to search for {DATA_TYPE} data: ")
+    deviceName = "Normal #1"  # input()
+    deviceCount = 0
+    for device in c8y.device_inventory.select(type="c8y_EventBasedSimulator", name=deviceName):
+        print(f"Found device '{device.name}', id: #{device.id}, "
+              f"owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        deviceCount += 1
+        print(f"List of {device.name}'s child devices: ")
+        for childDevice in device.child_devices:
+            print(f"Child device {childDevice.name}, id #{childDevice.id}")
+            if DATA_TYPE == "alarms":
+                listAlarms(c8y, childDevice, createFrom, createTo, filePath)
+            elif DATA_TYPE == "measurements":
+                # listing measurements of child device
+                listMeasurements(c8y, childDevice, createFrom, createTo, filePath)
+
+    if deviceCount == 0:
+        print(f"No device with name {deviceName} found")
+
+
+def listAlarms(c8y, childDevice, createFrom, createTo, filePath):
+    count = 0
+    for alarm in c8y.alarms.select(source=childDevice.id, created_after=createFrom, created_before=createTo):
+        print(
+            f"Found alarm id #{alarm.id}, severity: {alarm.severity}, time: {alarm.time}, creation time: {alarm.creation_time}, update time : {alarm.updated_time}\n")
+        print(f"{alarm.text}\n")
+        print(f"fragments: {list(alarm.keys())}\n")
```

Comment:
- no need to print all of this to the console; otherwise the execution time is just to long ...